### ISSUE TITLE
standardize_weights doesn't work with return_sequences=True and sample weights

### ISSUE THREAD

**Issue by cmyr** [fchollet/keras#1537]
as of the backend change, `standardize_weights` doesn't work correctly with `return_sequences=True`.

assuming I have a mini-batch size of 1600, and my `y.shape == (1600, 20, 1000)`, and my `weights.shape == (1600, 20, 1)` if I call `standardize_weights(y, None, None)` I get weights with shape `(1600,)`, but If I call  `standardize_weights(y, weights, None)` I get weights with shape `(32000,)`, e.g. my weights are just flattened and returned.

What is the purpose of the flatten call?

**Comment by fchollet** [comment]
`sample_weights` are for samples, and therefore there should be a 1:1 mapping between samples and weights. You cannot pass 3D weights or 4D weights, they must always be 2D.

**Comment by cmyr** [comment]
@fchollet right, but I have 20 samples / time-steps per input. Should I not be able to weight specific time-steps?

**Comment by fchollet** [comment]
No, a "sample" in the context of 3D time data would be a 2D sequence.

**Comment by cmyr** [comment]
So this means that the weights are applied equally to each item in the sequence? In earlier versions of keras I was using sample_weights to mask padding values in my sequences. It looked like it was working. Can you think of another way I can replicate this functionality?

**Comment by fchollet** [comment]
You can incorporate your timestep weights (!= sample weights) into your targets (`y`) and write a custom loss function...

**Comment by cmyr** [comment]
@fchollet okay thanks I'll give that some thought.

**Comment by DingKe** [comment]
So, I think the [document](http://keras.io/models/) is  outdated which says 
>For time-distributed data, there is one weight per sample per timestep, i.e. if your output data is shaped (nb_samples, timesteps, output_dim), your mask should be of shape (nb_samples, timesteps, 1). This allows you to mask out or reweight individual output timesteps, which is useful in sequence to sequence learning."

**Comment by fchollet** [comment]
Right. I just updated the docs.
